### PR TITLE
Add logger to FollowerState

### PR DESCRIPTION
**Pull request by janezpodhostnik**
Extracting pieces from https://github.com/onflow/flow-go/pull/3736 so its easier to digest and merge.

This one adds a currently unused log to follower state, to be used in the next PR.

### CODE REVIEW

**Review by jordanschalm — APPROVED** [review]

**Review by durkmurder — APPROVED** [review]
The only thing I would suggest is to change logger to be the first argument, in many places we try to pass it as first argument, using automatic refactoring it could be done very easily.

File: consensus/integration/nodes_test.go [hunk 1]
```diff
@@ -377,6 +377,14 @@ func createNode(
 	statusesDB := storage.NewEpochStatuses(metricsCollector, db)
 	consumer := events.NewDistributor()
 
+	localID := identity.ID()
+
+	// log with node index an ID
```
Comment (pattyshack) [inline consensus/integration/nodes_test.go:382]: rm comment

**Review by pattyshack — APPROVED** [review]

**Comment by janezpodhostnik** [comment]
@durkmurder I feel like the tracer and the logger belong together contextually. I think it would make sense moving the tracer as well then:

```
// state. This implementation is suitable only for NON-Consensus nodes.
func NewFollowerState(
	logger zerolog.Logger,
	tracer module.Tracer,
	state *State,
	index storage.Index,
	payloads storage.Payloads,
	consumer protocol.Consumer,
	blockTimer protocol.BlockTimer,
) (*FollowerState, error) 
```
what do you think?

**Comment by durkmurder** [comment]
> @durkmurder I feel like the tracer and the logger belong together contextually. I think it would make sense moving the tracer as well then:
> 
> 
> 
> ```
> 
> // state. This implementation is suitable only for NON-Consensus nodes.
> 
> func NewFollowerState(
> 
> 	logger zerolog.Logger,
> 
> 	tracer module.Tracer,
> 
> 	state *State,
> 
> 	index storage.Index,
> 
> 	payloads storage.Payloads,
> 
> 	consumer protocol.Consumer,
> 
> 	blockTimer protocol.BlockTimer,
> 
> ) (*FollowerState, error) 
> 
> ```
> 
> what do you think?

Yup, you could also move consumer as well, also attaching [example](https://github.com/onflow/flow-go/blob/master/engine/consensus/compliance/core.go) where logger, consumers, metrics, trading are grouped together and later other dependencies

**Review by durkmurder — APPROVED** [review]
Nice

**Comment by janezpodhostnik** [comment]
bors merge

**Pull request merged**